lib/recurly/client.rb

```diff
@@ -139,25 +194,14 @@ def handle_response!(request, faraday_resp)
       resource
     end
 
-    def raise_network_error!(ex)
-      error_class = case ex
-        when Faraday::TimeoutError
-          Errors::TimeoutError
-        when Faraday::ConnectionFailed
-          Errors::ConnectionFailedError
-        when Faraday::SSLError
-          Errors::SSLError
-        else
-          Errors::NetworkError
-        end
-
-      raise error_class, ex.message
-    end
-
     def raise_api_error!(response)
-      error = JSONParser.parse(self, response.body)
-      error_class = Errors::APIError.error_class(error.type)
-      raise error_class.new(response, error)
+      if response.content_type.include?(JSON_CONTENT_TYPE)
+        error = JSONParser.parse(self, response.body)
+        error_class = Errors::APIError.error_class(error.type)
+        raise error_class.new(response, error)
+      else
+        raise Recurly::Errors::InvalidResponseError, "Unexpected content type: #{http_response.content_type}"
```

Comment:
I believe this line needs to be changed to as `http_response` doesn't exist in this context:

```ruby
raise Recurly::Errors::InvalidResponseError, "Unexpected content type: #{response.content_type}"
```

Reply:
Thanks, my mistake as I refactored to reduce complexity. I added some specs around these conditions to better flush this out.